golos.vesting/golos.vesting.hpp

```diff
@@ -39,15 +43,30 @@ class vesting : public eosio::contract {
     void calculate_delegate_vesting();
 };
 
-asset vesting::get_account_vesting(account_name account, symbol_type sym)const {
+asset vesting::get_account_vesting(account_name account, symbol_type sym)const {//TODO: ?change to symbol_name sym
     tables::account_table balances(_self, account);
-    auto balance = balances.find(sym.name());
+    const auto& balance = balances.get(sym.name());
+    eosio_assert(balance.vesting.symbol == sym, "vesting::get_account_vesting: symbol precision mismatch" );
+    return balance.vesting;
+};
 
-    if (balance != balances.end())
-        return balance->vesting;
+asset vesting::get_account_available_vesting(account_name account, symbol_name sym)const {
+    tables::account_table balances(_self, account);
+    const auto& balance = balances.get(sym);
+    return (balance.vesting - balance.delegate_vesting);
+};
 
-    return asset(0, sym);
+asset vesting::get_account_sum_vesting(account_name account, symbol_name sym)const {
```

Comment:
please, rename it to `get_account_effective_vesting` (to conform with Golos). actually we can add `effective_vesting`, `available_vesting` functions to balance struct, this will allow to have logic in 1 place and reuse `available_vesting` in `sub_balance` assert

Reply:
done